pkg/controller/useraccount/useraccount_controller_test.go

```diff
@@ -1133,6 +1220,164 @@ func TestDisabledUserAccount(t *testing.T) {
 	})
 }
 
+func TestLookupAndDeleteCheUser(t *testing.T) {
+	// given
+	username := "sugar"
+	userID := uuid.NewV4().String()
+
+	t.Run("che user deletion is not enabled", func(t *testing.T) {
+		// given
+		userAcc := newUserAccount(username, userID, false)
+		r, _, _ := prepareReconcile(t, username, userAcc, cheRoute(true), keycloackRoute(true))
+
+		// when
+		err := r.lookupAndDeleteCheUser(userAcc)
+
+		// then
+		require.NoError(t, err)
+	})
+
+	t.Run("che user deletion is enabled", func(t *testing.T) {
+		memberOperatorSecret := newSecretWithCheAdminCreds()
+		restore := test.SetEnvVarsAndRestore(t,
+			test.Env("WATCH_NAMESPACE", "toolchain-member"),
+			test.Env("MEMBER_OPERATOR_SECRET_NAME", "test-secret"),
+			// Che has two different routes for Che & Keycloak but CRW uses a single route for both.
+			// For tests we'll assume they're two different routes.
+			test.Env(configuration.MemberEnvPrefix+"_CHE_KEYCLOAK_ROUTE_NAME", "keycloak"),
+		)
+		defer restore()
+
+		t.Run("get token error", func(t *testing.T) {
+			// given
+			defer gock.OffAll()
+			gockTokenFail()
+			gockFindUserSuccess(username)
+			gockDeleteUser(204)
+			userAcc := newUserAccount(username, userID, false)
+			r, _, _ := prepareReconcile(t, username, userAcc, memberOperatorSecret, cheRoute(true), keycloackRoute(true))
+
+			//when
+			err := r.lookupAndDeleteCheUser(userAcc)
+
+			// then
+			require.EqualError(t, err, `request to find Che user 'sugar' failed: unable to obtain access token for che, Response status: '400 Bad Request'. Response body: ''`)
+			userAcc = &toolchainv1alpha1.UserAccount{}
+			err = r.client.Get(context.TODO(), types.NamespacedName{Name: username, Namespace: test.MemberOperatorNs}, userAcc)
+			require.NoError(t, err)
+		})
+
+		t.Run("user not found", func(t *testing.T) {
+			// given
+			defer gock.OffAll()
+			gockTokenSuccess()
+			gockFindUserNoBody(username, 404)
+			gockDeleteUser(204)
```

Comment:
if the user wasn't found, then you don't need this mock
```suggestion
```

Reply:
or you could return an error:
```suggestion
			gockDeleteUser(404)
```

Reply:
Done